### ISSUE TITLE
Suggest using openpgp4fpr URI scheme for Encryption

### ISSUE THREAD

**Issue by wiktor-k** [securitytxt/security-txt#107]
I've noticed that the `Encryption` field can take any URI. Maybe it would be useful to add reference to `openpgp4fpr` URIs that reference OpenPGP key by fingerprint. These URIs are [used by several applications](https://github.com/effigies/openpgp4info#about-openpgp4fpr) already.

Using `openpgp4fpr` URI over `https` reference to the key has the benefit that the aiuthoritative key reference is embedded wholly in the URI.

For example if security.txt contained an `https` reference: `Encryption: https://example.com/key.asc` and there was `signature.txt.sig` file then the attacker could easily replace `key.asc` and the signature would still be valid while the key would be fake. On the other hand if security.txt contained `Encryption: openpgp4fpr:653909A2F0E37C106F5FAF546C8857E0D8E8F074` then it's computationally expensive to create a key with the same fingerprint.

**Comment by nightwatchcyber** [comment]
Is there an RFC or an IANA registration for "openpgp4fpr"?

**Comment by wiktor-k** [comment]
@nightwatchcyber nope, not at the moment, but it's a good idea to register at least a provisional URI scheme, I'll work on that and update this ticket when new info is available.

**Comment by jeroenh** [comment]
Already in the works it seems: https://datatracker.ietf.org/doc/draft-openpgp-iana-registry-updates/

**Comment by jeroenh** [comment]
You’re right, I hit “send” too soon. It may be worth reaching out to the authors of that draft to discuss adding that to the draft or whether it has been discussed before?

Given that it is already adopted by W3’s WHATWG, it seems like an easy case to make.

**Comment by nightwatchcyber** [comment]
We really need some sort of a spec or reference to a document where the scheme is defined. For WHATWG, I was only able to find a reference to the scheme itself but not it's definition:
https://html.spec.whatwg.org/multipage/system-state.html#safelisted-scheme

Maybe someone can reach out to @effigies to get additional information. All I can find is a mailing list message:
http://permalink.gmane.org/gmane.comp.security.monkeysphere/592

**Comment by effigies** [comment]
Hi all, this is basically all the information I have: https://github.com/effigies/openpgp4info/blob/master/README.md

The schema is `openpgp4fpr:<fingerprint>` where `<fingerprint>` is 40 hex characters. There's not really much else to it, as far as I can tell.

**Comment by wiktor-k** [comment]
Yes, that's that simple :)
IANA added the entry to schemes registry: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

**Comment by effigies** [comment]
Oh, ha. You already linked to that page. Yeah, sorry I can't be of more help. I don't have any arcane knowledge here.

**Comment by effigies** [comment]
The discussion here may be of interest to you, as well: open-keychain/open-keychain#1281

**Comment by effigies** [comment]
> I plan to run the protocol handler on my domain too.

Cool. Drop me a line when you do, and I can add it to the README.

**Comment by wiktor-k** [comment]
I don't see an updated version here https://tools.ietf.org/html/draft-foudil-securitytxt-03 but since the change was merged maybe this issue can be closed?

@effigies I sent you an email a long time ago...
